### PR TITLE
Add function to fill a buffer with random bytes

### PR DESCRIPTION
**Pull request by pyohannes**
This PR adds a method `GenerateRandomBuffer`, which fills a buffer with random bytes. This will be useful for span and trace id creation, which are represented as arrays of `uint8_t`s.

Furthermore, this adds a slim interface around the number generator, encapsulating the actual generator used and exposing the static methods `GenerateRandom64` and `GenerateRandomBuffer`.

### CODE REVIEW

File: sdk/test/common/random_test.cc [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
 
-TEST(RandomTest, GenerateRandomNumbers)
+TEST(RandomTest, GenerateRandomBuffer)
 {
-  auto &random_number_generator = GetRandomNumberGenerator();
-  EXPECT_NE(random_number_generator(), random_number_generator());
+  uint8_t buf1[8];
+  uint8_t buf2[8];
+  Random::GenerateRandomBuffer(buf1);
+  Random::GenerateRandomBuffer(buf2);
```
Comment (g-easy) [inline sdk/test/common/random_test.cc:19]: IMO test some edge cases as well
7, 8, 9, 16 ?

**Comment by reyang** [comment]
@pyohannes is this ready for merge? If yes, please rebase, thanks!

**Comment by pyohannes** [comment]
@reyang This is now rebased and ready to merge.
